src/App.tsx

```diff
@@ -16,104 +11,90 @@ function MidiToNotes (midiNotes: {[key: number]: Note}) : Notes {
   return notes;
 }
 
-type AppState = {
-  access: Promise<WebMidi.MIDIAccess>,
-  webMidiStatus: WebMidiStatus,
-  selected_input?: string,
-  pressedKeys: {[key: number]: Note}
-};
-
-class App extends React.Component<EmptyProps, AppState> {
-  _mounted: boolean;
+const App = () => {
 
-  constructor(props: EmptyProps) {
-    super(props);
+  const [state, setState] = useState<AppState>({
+    access: navigator.requestMIDIAccess(),
+    webMidiStatus: "initializing",
+    pressedKeys: {},
+  })
 
-    // Used to detect a potential race condition between WebMidi and React while mounting the component
-    this._mounted = false;
-
-    WebMidi.enable((err) => {
+  useEffect(() => {
+    WebMidi.enable((err) => { 
       if (err) {
         console.log("WebMidi could not be enabled.", err);
-        if (this._mounted) {
-          this.setState(() => { return { webMidiStatus: "error" }; });
-        } else {
-          this.state = Object.assign(this.state, { webMidiStatus: "error" });
-        }
+        setState({
+          ...state,
+          webMidiStatus: "error"
+        });
```

Comment:
Maybe we can use an [updater](https://stackoverflow.com/a/43639228) here?

```suggestion
        setState((prevState) => {
          ...prevState,
          webMidiStatus: "error"
        });
```